src/wp-includes/media.php

```diff
@@ -5640,6 +5670,20 @@ function wp_get_loading_optimization_attributes( $tag_name, $attr, $context ) {
 		return $loading_attrs;
 	}
 
+	/*
+	 * Add `decoding` with a value of "async" for every image unless it has a
+	 * conflicting `decoding` attribute already present.
+	 */
+	if ( 'img' === $tag_name ) {
```

Comment:
Please see my previous feedback in https://github.com/WordPress/wordpress-develop/pull/4991#discussion_r1329151645: Right now, this will add `decoding` attributes even when in one of the special contexts in which it shouldn't make any modifications (the clause in line 5700). As mentioned, please move that clause above this one here so that it still applies first.